cmd/syncthing/gui.go

```diff
@@ -398,15 +414,20 @@ func (s *apiService) String() string {
 }
 
 func (s *apiService) VerifyConfiguration(from, to config.Configuration) error {
-	_, err := net.ResolveTCPAddr("tcp", to.GUI.Address())
-	return err
+	for _, addr := range to.GUI.Addresses() {
+		_, err := net.ResolveTCPAddr("tcp", addr)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func (s *apiService) CommitConfiguration(from, to config.Configuration) bool {
 	// No action required when this changes, so mask the fact that it changed at all.
 	from.GUI.Debugging = to.GUI.Debugging
 
-	if to.GUI == from.GUI {
+	if reflect.DeepEqual(to.GUI, from.GUI) {
 		return true
 	}
```

Comment:
Why does this need a reflect?

Reply:
Because the GUI configuration now has a slice field which is a reference type and does not support native structural equality.

Reply:
It's sad but true :(